src/sync.js

```diff
@@ -701,7 +701,31 @@ SyncApi.prototype._syncFromCache = async function(savedSync) {
     };
 
     try {
-        await this._processSyncResponse(syncEventData, data);
+        await this._processSyncResponse(syncEventData, data, false);
+
+        const breadcrumbs = this.client.store.getAccountData("im.vector.setting.breadcrumbs");
```

Comment:
Breadcrumbs are an Element Web specific feature that JS SDK shouldn't have to concern itself with directly, so we may need some option / callback / hook that handles this a bit differently. Perhaps there's an optional function the app can pass to SDK which gives back a generic array of "eagerly decrypted rooms"...? And then in React SDK, we would implement that to look at the breadcrumbs.

Reply:
Or instead, maybe the React SDK just does this work itself after some `sync` event is emitted from JS SDK, since none of this seems specific to the data being manipulated as part of the sync action.